lib/ready.js

```diff
@@ -80,6 +80,9 @@ class Ready extends EventEmitter {
     opt.name = name || cacheKey;
     const timer = setTimeout(() => this.emit('ready_timeout', opt.name), opt.timeout);
     const cb = once(err => {
+      if (err != null && !(err instanceof Error)) {
+        err = new Error(err);
```

Comment:
How about move this logic to readyDone, so you can deal with error together at one place.